pkg/providers/tinkerbell/controlplane.go

```diff
@@ -0,0 +1,120 @@
+package tinkerbell
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	tinkerbellv1 "github.com/tinkerbell/cluster-api-provider-tinkerbell/api/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/aws/eks-anywhere/pkg/clients/kubernetes"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/clusterapi"
+	yamlcapi "github.com/aws/eks-anywhere/pkg/clusterapi/yaml"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/yamlutil"
+)
+
+// BaseControlPlane represents a CAPI Tinkerbell control plane.
+type BaseControlPlane = clusterapi.ControlPlane[*tinkerbellv1.TinkerbellCluster, *tinkerbellv1.TinkerbellMachineTemplate]
+
+// ControlPlane holds the Tinkerbell specific objects for a CAPI Tinkerbell control plane.
+type ControlPlane struct {
+	BaseControlPlane
+	Secrets *corev1.Secret
+}
+
+// Objects returns the control plane objects associated with the Tinkerbell cluster.
+func (p ControlPlane) Objects() []kubernetes.Object {
+	o := p.BaseControlPlane.Objects()
+	o = append(o, p.Secrets)
+
+	return o
+}
+
+// ControlPlaneSpec builds a Tinkerbell ControlPlane definition based on an eks-a cluster spec.
+func ControlPlaneSpec(ctx context.Context, logger logr.Logger, client kubernetes.Client, clusterSpec *cluster.Spec) (*ControlPlane, error) {
+	templateBuilder, err := generateTemplateBuilder(clusterSpec)
+	if err != nil {
+		return nil, errors.Wrap(err, "generating tinkerbell template builder")
+	}
+
+	controlPlaneYaml, err := templateBuilder.GenerateCAPISpecControlPlane(
+		clusterSpec,
+		func(values map[string]interface{}) {
+			values["controlPlaneTemplateName"] = clusterapi.ControlPlaneMachineTemplateName(clusterSpec.Cluster)
+			values["etcdTemplateName"] = clusterapi.EtcdMachineTemplateName(clusterSpec.Cluster)
```

Comment:
We don't support external etcd so I suspect we can remove this.

Reply:
Shall we remove this logic for CLI as well? https://github.com/aws/eks-anywhere/blob/main/pkg/providers/tinkerbell/tinkerbell.go#L130